probe-rs/src/probe/blackmagic/arm.rs

```diff
@@ -0,0 +1,815 @@
+
+                CSW::try_from(Into::<u32>::into(csw))?
+            }
+            MemoryAp::AmbaApb4Apb5(ap) => {
+                let mut csw = ap.status(self)?;
+
+                csw.DbgSwEnable = true;
+                csw.AddrInc = AddressIncrement::Off;
+                csw.Size = DataSize::U8;
+
+                // Enable secure access if it's allowed
+                csw.NonSecure = !csw.SPIDEN;
+                csw.Privileged = true;
+
+                CSW::try_from(Into::<u32>::into(csw))?
+            }
+            MemoryAp::AmbaAxi3Axi4(ap) => {
+                let mut csw = ap.status(self)?;
+
+                csw.DbgSwEnable = true;
+                csw.AddrInc = AddressIncrement::Off;
+                csw.Size = DataSize::U8;
+
+                csw.Instruction = false;
+                // Enable secure access if it's allowed
+                csw.NonSecure = !csw.SPIDEN;
+                csw.Privileged = true;
+                csw.CACHE = 0;
+
+                CSW::try_from(Into::<u32>::into(csw))?
+            }
+            MemoryAp::AmbaAxi5(ap) => {
+                let mut csw = ap.status(self)?;
+
+                csw.DbgSwEnable = true;
+                csw.AddrInc = AddressIncrement::Off;
+                csw.Size = DataSize::U8;
+
+                csw.Instruction = false;
+                // Enable secure access if it's allowed
+                csw.NonSecure = !csw.SPIDEN;
+                csw.Privileged = true;
+                csw.CACHE = 0;
+                csw.MTE = false;
+
+                CSW::try_from(Into::<u32>::into(csw))?
+            }
+        };
+
+        Ok(Box::new(BlackMagicProbeMemoryInterface {
+            probe: self,
+            current_ap,
+            index: 0,
+            apsel: 0,
+            csw: csw.into(),
+        }) as _)
+    }
+
+    fn read_chip_info_from_rom_table(
+        &mut self,
+        dp: DpAddress,
+    ) -> Result<Option<crate::architecture::arm::ArmChipInfo>, ArmError> {
+        if dp != DpAddress::Default {
+            return Err(ArmError::NotImplemented("multidrop not yet implemented"));
+        }
+
+        for ap in self.access_ports.clone() {
+            if let Ok(mut memory) = self.memory_interface(&ap) {
+                let base_address = memory.base_address()?;
+                let component = Component::try_parse(&mut *memory, base_address)?;
+
+                if let Component::Class1RomTable(component_id, _) = component {
+                    if let Some(jep106) = component_id.peripheral_id().jep106() {
+                        return Ok(Some(ArmChipInfo {
+                            manufacturer: jep106,
+                            part: component_id.peripheral_id().part(),
+                        }));
+                    }
+                }
+            }
+        }
+
+        Ok(None)
+    }
+}
+
+impl SwoAccess for BlackMagicProbeArmDebug {
+    fn enable_swo(&mut self, config: &crate::architecture::arm::SwoConfig) -> Result<(), ArmError> {
```

Comment:
All of the trait methods should return an error (instead of panicking) (there should already be a `NotImplemented` variant) and not print anything :)

Reply:
Also, is the implementation of this trait really required? It should be optional I believe.

Reply:
Yes, it's required because `ArmProbeInterface` requires it to be implemented.

Reply:
Hmm that's suboptimal considering the fact that the methods seemingly are never called 😅

Reply:
They should still return `NotImplemented` instead of `unimplemented!()` to provide a nicer user experience :)